api/test/unit/utils/tallies_test.py

```diff
@@ -0,0 +1,113 @@
+from datetime import datetime
+
+import pytest
+from freezegun import freeze_time
+
+from catalog.api.utils import tallies
+
+
+# Use a fake media type to avoid having to run each test case for each media type
+FAKE_MEDIA_TYPE = "this_is_not_a_media_type"
+
+
+@pytest.mark.parametrize(
+    ("now", "expected_timestamp"),
+    (
+        pytest.param(datetime(2023, 1, 19), "2023-01-16", id="midweek"),
+        pytest.param(datetime(2023, 1, 16), "2023-01-16", id="start_of_week"),
+        pytest.param(datetime(2023, 1, 15), "2023-01-09", id="end_of_week"),
+    ),
+)
+def test_count_provider_occurrences_uses_week_timestamp(now, expected_timestamp, redis):
+    results = [{"provider": "flickr"} for _ in range(4)] + [
+        {"provider": "stocksnap"} for _ in range(6)
+    ]
+    with freeze_time(now):
+        tallies.count_provider_occurrences(results, FAKE_MEDIA_TYPE)
+
+    assert (
+        redis.get(f"provider_occurrences:{FAKE_MEDIA_TYPE}:{expected_timestamp}:flickr")
+        == b"4"
+    )
+    assert (
+        redis.get(
+            f"provider_occurrences:{FAKE_MEDIA_TYPE}:{expected_timestamp}:stocksnap"
+        )
+        == b"6"
+    )
+    assert (
+        redis.get(
+            f"provider_appeared_in_searches:{FAKE_MEDIA_TYPE}:{expected_timestamp}:flickr"
+        )
+        == b"1"
+    )
+    assert (
+        redis.get(
+            f"provider_appeared_in_searches:{FAKE_MEDIA_TYPE}:{expected_timestamp}:stocksnap"
+        )
+        == b"1"
+    )
+
+
```

Comment:
Did we want to add a test (or would it be easily possible to test) that no results are incremented for page queries > 4?

Reply:
This is tested by the test that verifies that the function isn't called by `search_controller.search` if the page is 5. Is there a different approach you would like to take?